jobs/statistics/ProcessUsageStatsLogFile.php

```diff
@@ -0,0 +1,246 @@
+                error_log($message);
+            }
+            throw new JobException(__('admin.job.processLogFile.openFileFailed', ['file' => $dispatchFilePath]));
+        }
+
+        // Make sure we don't have any temporary records associated
+        // with the current load ID in database.
+        $this->deleteByLoadId();
+
+        $lineNumber = 0;
+        while (!feof($fhandle)) {
+            $lineNumber++;
+            $line = trim(fgets($fhandle));
+            if (empty($line) || substr($line, 0, 1) === '#') {
+                continue;
+            } // Spacing or comment lines. This actually should not occur in the new format.
+
+            $entryData = json_decode($line);
+            if ($entryData === null) {
+                // This line is not in the right format.
+                $message = __(
+                    'admin.job.processLogFile.wrongLoglineFormat',
+                    ['file' => $this->loadId, 'lineNumber' => $lineNumber]
+                );
+                error_log($message);
+                continue;
+            }
+
+            try {
+                $this->isLogEntryValid($entryData);
+            } catch (Exception $e) {
+                $message = __(
+                    'admin.job.processLogFile.invalidLogEntry',
+                    ['file' => $this->loadId, 'lineNumber' => $lineNumber, 'error' => $e->getMessage()]
+                );
+                error_log($message);
+                continue;
+            }
+
+            // Avoid bots.
+            if (Core::isUserAgentBot($entryData->userAgent)) {
+                continue;
+            }
+
+            $this->insertTemporaryUsageStatsData($entryData, $lineNumber);
+        }
+        fclose($fhandle);
+    }
+
+    /**
+     * Delete entries in usage stats temporary tables by loadId
+     */
+    protected function deleteByLoadId(): void
+    {
+        $temporaryInstitutionsDao = DAORegistry::getDAO('TemporaryInstitutionsDAO'); /** @var TemporaryInstitutionsDAO $temporaryInstitutionsDao */
+        $temporaryTotalsDao = DAORegistry::getDAO('TemporaryTotalsDAO'); /** @var TemporaryTotalsDAO $temporaryTotalsDao */
+        $temporaryItemInvestigationsDao = DAORegistry::getDAO('TemporaryItemInvestigationsDAO'); /** @var TemporaryItemInvestigationsDAO $temporaryItemInvestigationsDao */
+        $temporaryItemRequestsDao = DAORegistry::getDAO('TemporaryItemRequestsDAO'); /** @var TemporaryItemRequestsDAO $temporaryItemRequestsDao */
+
+        $temporaryInstitutionsDao->deleteByLoadId($this->loadId);
+        $temporaryTotalsDao->deleteByLoadId($this->loadId);
+        $temporaryItemInvestigationsDao->deleteByLoadId($this->loadId);
+        $temporaryItemRequestsDao->deleteByLoadId($this->loadId);
+    }
+
+    /**
+     * Validate the usage stats log entry
+     *
+     * @throws Exception.
+     */
+    protected function isLogEntryValid(object $entry): void
```

Comment:
As it's not returning a boolean, I think it's better to rename it to `validateLogEntry()` or something like that.